identity/keystone.go

```diff
@@ -63,7 +64,7 @@ func keystoneAuth(client *goosehttp.Client, auth_data interface{}, URL string) (
 	requestData := goosehttp.RequestData{ReqValue: auth_data, RespValue: &accessWrapper}
 	err := client.JsonRequest("POST", URL, "", &requestData, nil)
 	if err != nil {
-		return nil, err
+		return nil, gooseerrors.Newf(err, "", "requesting token failed")
```

Comment:
No need for the "". The args are standard string formatting args, a format string followed by param values.
You just want gooseerrors.Newf(err, "requesting token failed")